common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1572,3 +1572,11 @@ def set_user_partitions_for_scheme(self, partitions, scheme):
             if p.scheme != scheme
         ]
         self.user_partitions = other_partitions + partitions  # pylint: disable=attribute-defined-outside-init
+
+    def is_discussion_enabled(self, user):
+        """
+        Return True if Discussion is enabled for a course; else False
+        """
+        if not any([tab.type == 'discussion' and tab.is_enabled(self, user) for tab in self.tabs]):
```

Comment:
Also, for performance reasons, you'll want to pass in `None` for the `user` to `DiscussionTab.is_enabled` since the mobile_api's Enrollment endpoint already checks the user's permissions before that.